src/mscorlib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -78,12 +78,22 @@ public override unsafe int GetByteCount(char* chars, int count)
             return EncodingForwarder.GetByteCount(this, chars, count);
         }
 
-        public override int GetBytes(String chars, int charIndex, int charCount,
-                                              byte[] bytes, int byteIndex)
-        {
-            return EncodingForwarder.GetBytes(this, chars, charIndex, charCount, bytes, byteIndex);
+#if !BIGENDIAN
```

Comment:
This is acceptable for now to have the ifdef as we have other code placed under same ifdef too. 
@jkotas what you think of changing this to runtime check instead of compile time check? I mean we can detect if we are running on BIGENDIA.